pkg/schema/error.go

```diff
@@ -0,0 +1,158 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package schema
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/k14s/ytt/pkg/filepos"
+	"github.com/k14s/ytt/pkg/yamlmeta"
+)
+
+type errorWithContent struct {
+	contents string
+	Position *filepos.Position
+}
+
+func (e errorWithContent) Error() string {
+	panic("do not use directly")
+}
+
+func (e *errorWithContent) SetContext(doc *yamlmeta.Document) error {
+	contents, err := doc.RawDataAtLine(e.Position)
+
+	if err == nil {
+		e.contents = contents
+	}
+
+	return err
+}
+
+func NewInvalidError(found, expected, hint string, position *filepos.Position) error {
+	return &invalidError{
+		errorWithContent: errorWithContent{
+			Position: position,
+		},
+		Found:    found,
+		Expected: expected,
+		Hint:     hint,
+	}
+}
+
+type invalidError struct {
```

Comment:
"invlaidError" makes me think that the error is invalid, not that the schema is invalid.

Reply:
👍 Agreed. Renamed to invalidSchemaError